src/main/java/seedu/address/model/ReadOnlyModel.java

```diff
@@ -63,6 +66,9 @@
     /** Returns the filtered deadline task list as an unmodifiable ObservableList */
     ObservableList<IndexedItem<DeadlineTask>> getDeadlineTaskList();
 
+    /** Return a filtered Deadline task list filter by given task predicate. */
+    ObservableList<DeadlineTask> getDeadlineTaskList(TaskPredicate taskFilter);
+
```

Comment:
Also, this comment is wrong. This returns the underlying *unfiltered* task list.